source/extensions/filters/http/istio_stats/istio_stats.cc

```diff
@@ -1107,9 +1107,19 @@ class IstioStatsFilter : public Http::PassThroughFilter,
     case Reporter::ClientSidecar: {
       const Ssl::ConnectionInfoConstSharedPtr ssl_info =
           info.upstreamInfo() ? info.upstreamInfo()->upstreamSslConnection() : nullptr;
+      std::optional<Istio::Common::WorkloadMetadataObject> endpoint_peer;
       if (ssl_info && !ssl_info->uriSanPeerCertificate().empty()) {
         peer_san = ssl_info->uriSanPeerCertificate()[0];
       }
+      if (peer_san.empty()) {
+        auto endpoint_object = peerInfo(config_->reporter(), filter_state);
```

Comment:
can we reuse L1016?

Reply:
Oh wow thought I responded to this. No we can't reuse because of scoping and the nested switches